infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/environment/PodMerger.java

```diff
@@ -154,8 +155,13 @@ public Deployment merge(List<PodData> podsData) throws ValidationException {
 
       // if there are entries with such keys then values will be overridden
       baseSpec.getAdditionalProperties().putAll(podData.getSpec().getAdditionalProperties());
+
       // add tolerations to baseSpec if any
-      baseSpec.getTolerations().addAll(podData.getSpec().getTolerations());
+      for (Toleration toleration : podData.getSpec().getTolerations()) {
+        if (!baseSpec.getTolerations().contains(toleration)) {
+          baseSpec.getTolerations().add(toleration);
```

Comment:
Is this the expected way this would work, or should trying to add the same toleration twice result in an error? From looking at the test, it looks like the second toleration would just get ignored.

Reply:
@amisevsk The problem is that this code is embedded inside a loop. The other attributes get around this issue by calling their respective 'merge...' methods, but there wasn't one for Tolerations and the logic was so simple that I wasn't sure it was worth making one.
But to get to the heart of your question, we were using this with a correct configuration in values.yaml that was only specifying the Toleration once. This code was responsible for the duplication. So it wasn't a case of user error that should cause any kind of failure / error.